### PR TITLE
[5.1] Strengthen Security

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes

- Disable PHP execution inside the `media` folder
- Option to apply an `.htaccess` to each filesystem-local folder (defaults to false for b/c)

### Testing Instructions

- Apply the patch.
- create a file `media/test.php` with contents `<?= phpinfo();`
- try to access the file, ie localhost/media/test.php (you shouldn't get the phpinfo page)
- Goto `/administrator/index.php?option=com_plugins&view=plugins` and edit the filesystem local plugin by enabling the `Strengthen Security` switch
- Create a file `images/test.php` with contents `<?= phpinfo();`
- try to access the file, ie localhost/images/test.php (you shouldn't get the phpinfo page)
- Goto `/administrator/index.php?option=com_plugins&view=plugins` and edit the filesystem local plugin by disabling the `Strengthen Security` switch
- check that the images folder doesn't contain an `.htaccess` file

### Actual result BEFORE applying this Pull Request

PHP execution inside the static folders

### Expected result AFTER applying this Pull Request

PHP execution is prevented inside the static folders

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by brianteeman** [comment]
I am not a fan of this approach although I do agree with its intentions.

Instead of allowing php to be executable in every folder and then adding a deny file in every media folder I would simply suggest adding a deny all in the root htaccess and then people can add exceptions to that rule.

For the core there are only a few exceptions

**Comment by dgrammatiko** [comment]
@brianteeman I totally agree that it's better to have secure by default ootb. The only problems are:
- that it cannot happen in a patch version (if I'm not wrong)
- it will impact heavily some extensions (and by extension end users)

thus this middle ground PR. Without enabling the new option only the `media` folder gets the php exclusion, which should have zero effects, as that folder should never have php files.
But if the maintainers feel that going full blast is ok for a patch version then someone with better apache knowledge should take over

**Comment by brianteeman** [comment]
> @brianteeman I totally agree that it's better to have secure by default ootb. The only problems are:
> 
> * that it cannot happen in a patch version (if I'm not wrong)

See comment below

> * it will impact heavily some extensions (and by extension end users)

If there really are extensions that allow direct access to any php files anywhere then they are cr*p and I dont care about them. There are very very very few legitimate cases for direct entry points such as the one in core for joomlaupdate

> 
> thus this middle ground PR. Without enabling the new option only the `media` folder gets the php exclusion, which should have zero effects, as that folder should never have php files. But if the maintainers feel that going full blast is ok for a patch version then someone with better apache knowledge should take over

If there are extensions abusing the non-media folders then there is no reason to assume that they are not also abusing the media folder.

I really dont see this as being anything more complex than previous changes to the core htaccess file which have always been for new installs only plus a post-installation message for existing sites. Exactly the same as seen in this PR #30221

File: plugins/filesystem/local/local.xml [hunk 1]
```diff
@@ -55,6 +55,17 @@
 							<option value="0">JNO</option>
 							<option value="1">JYES</option>
 						</field>
+						<field
+							name="strengthened"
+							type="radio"
+							label="PLG_FILESYSTEM_LOCAL_DIRECTORIES_DIRECTORY_STRENGTHENED_LABEL"
```
Comment (Quy) [inline plugins/filesystem/local/local.xml:61]: Add inline help to explain feature?

**Pull request closed without merging**